src/com/nutrons/FRamework/Robot.java

```diff
@@ -1,48 +1,68 @@
 package com.nutrons.FRamework;
 
-import static com.nutrons.FRamework.CompMode.AUTO;
-import static com.nutrons.FRamework.CompMode.TELEOP;
-import static com.nutrons.FRamework.CompMode.TEST;
-
 import edu.wpi.first.wpilibj.RobotBase;
 import io.reactivex.Flowable;
 
+import static com.nutrons.FRamework.CompMode.*;
+
 public class Robot extends RobotBase {
 
-  private Flowable<Boolean> enabled;
-  private Flowable<CompMode> mode;
+  private StreamManager sm;
 
   /**
-   * Initialize the robot.
+   * Bootstraps the StreamManager with appropriate factories.
    */
   public Robot() {
-    this.enabled = Util.changedValues(Util.toFlow(this::isEnabled));
-    this.mode = Util.changedValues(Flowable.merge(
-        Util.toFlow(this::isAutonomous).filter(x -> x).map((x) -> AUTO),
-        Util.toFlow(this::isOperatorControl).filter(x -> x).map((x) -> TELEOP),
-        Util.toFlow(this::isTest).filter(x -> x).map((x) -> TEST)));
+
   }
 
   /**
    * Called at start of competition.
    */
   @Override
-  public void startCompetition() {
-    this.enabled.ignoreElements().blockingAwait();
-    this.mode.ignoreElements().blockingAwait();
+  public final void startCompetition() {
+    this.sm = new StreamManager(
+        Util.changedValues(Util.toFlow(this::isEnabled)),
+        Util.changedValues(Flowable.merge(
+            Util.toFlow(this::isAutonomous).filter(x -> x).map((x) -> AUTO),
+            Util.toFlow(this::isOperatorControl).filter(x -> x).map((x) -> TELEOP),
+            Util.toFlow(this::isTest).filter(x -> x).map((x) -> TEST)))
+    );
+    this.sm.startCompetition();
   }
 
-  /**
-   * @return A Flowable representing the "enabled" state of the robot over time.
-   */
-  public Flowable<Boolean> enabled() {
-    return this.enabled;
+  @Override
+  public final void free() {
+    super.free();
   }
 
-  /**
-   * @return A Flowable representing the "competition mode" of the robot over time.
-   */
-  public Flowable<CompMode> mode() {
-    return this.mode;
+  @Override
```

Comment:
If you're just calling the superclass, do you need to override?  These methods don't seem to be abstract.